### PR TITLE
Migrated to Swift5

### PR DESCRIPTION
**Pull request by yysskk**
I fixed that it could not build on Swift5 due to dependencies :)

## Update dependencies
- RxSwift, ~> 5,  https://github.com/ReactiveX/RxSwift/releases
- Startscream, ~> 3.1, https://github.com/daltoniam/Starscream/releases

### CODE REVIEW

**Comment by freak4pc** [comment]
Thanks @yysskk - CI is failing

**Comment by Ricardo1980** [comment]
Works fine. I only had to modify
github "ReactiveX/RxSwift" ~> 5
to 
github "ReactiveX/RxSwift" ~> 5.0

Can we merge this PR?

File: .circleci/config.yml [hunk 1]
```diff
@@ -17,25 +17,19 @@ jobs:
...
-          command: set -o pipefail && xcodebuild test SWIFT_VERSION=4.2 -scheme RxStarscream -project RxStarscream.xcodeproj -sdk iphonesimulator -destination "name=iPhone 8" | xcpretty -c -r html --output $XCODE_TEST_REPORTS/iOS_swift4_2.html
-      - run:
-          name: macOS, Swift 4
-          command: set -o pipefail && xcodebuild test SWIFT_VERSION=4.0 -scheme RxStarscream-macOS -project RxStarscream.xcodeproj -sdk macosx | xcpretty -c -r html --output $XCODE_TEST_REPORTS/macOS_swift4.html
+          name: iOS, Swift 5
+          command: set -o pipefail && xcodebuild test SWIFT_VERSION=5.0 -scheme RxStarscream -project RxStarscream.xcodeproj -sdk iphonesimulator -destination "name=iPhone 8" |  xcpretty -c -r html --output $XCODE_TEST_REPORTS/iOS_swift5.html
       - run:
-          name: macOS, Swift 4.2
-          command: set -o pipefail && xcodebuild test SWIFT_VERSION=4.2 -scheme RxStarscream-macOS -project RxStarscream.xcodeproj -sdk macosx | xcpretty -c -r html --output $XCODE_TEST_REPORTS/macOS_swift4_2.html
+          name: macOS, Swift 5
```
Comment (freak4pc) [inline .circleci/config.yml:31]: Hey @yysskk - just one thing, there's no real reason to stop supporting Swift 4.2.
Xcode 10.2.1 supports both Swift 4.2 and 5, so does RxSwift 5 - I'd like this repo to keep the support & tests for both if possible. Could you do it?

**Review by freak4pc — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (kiyoshi-shikuma) [reply .circleci/config.yml:31]: @yysskk @freak4pc Any chance we can merge this now? I believe RxSwift 5 specifies Swift 5 as a requirement now, and Xcode 11.2 does not ship with Swift 4.2 support?

Comment (aeternas) [reply .circleci/config.yml:31]: Dear team,

Did you have any chance to make a decision whether we can merge this PR or not?

**Comment by kiyoshi-shikuma** [comment]
Can we please merge this? This is now somewhat urgent as RxSwift has eliminated the UIWebView dependency which will cause apps to get rejected starting in April per Apple's guidelines. AFAIK RxSwift 5.1 has now become a necessity.

**Comment by freak4pc** [comment]
There are some more changes needed here but I don't want to keep this PR hanging. Merging. 

If anyone is a heavy user of this framework and wants to manage this repo, please let me know. Thanks!

**Review by freak4pc — APPROVED** [review]

**Pull request merged**